src/en/manga18me/src/eu/kanade/tachiyomi/extension/en/manga18me/Manga18Me.kt

```diff
@@ -0,0 +1,219 @@
+package eu.kanade.tachiyomi.extension.en.manga18me
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.asObservableSuccess
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import rx.Observable
+import java.lang.Exception
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class Manga18Me : ParsedHttpSource() {
+
+    override val name = "Manga18Me"
+
+    override val lang = "en"
+
+    override val baseUrl = "https://manga18.me"
+
+    override val supportsLatest = true
+
+    override val client = network.client.newBuilder()
+        .rateLimit(3, 1)
+        .build()
+
+    override fun headersBuilder() = Headers.Builder().apply {
+        add("Referer", "$baseUrl/")
+    }
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/manga/$page?orderby=trending", headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val entries = document
+            .select(popularMangaSelector())
+            .filter { it ->
+                val searchText = document.selectFirst("div.section-heading h1")?.text() ?: ""
+                val raw = document.selectFirst("div.canonical")?.attr("href") ?: ""
+                val title = it.selectFirst("div.item-thumb.wleft a")?.attr("href") ?: ""
+
+                if (searchText.lowercase().contains("raw")) {
+                    true
+                } else if (raw.lowercase().contains("raw")) {
+                    true
+                } else if (title.contains("raw")) {
+                    false
+                } else {
+                    true
+                }
+            }

```

Comment:
what is this for?

Reply:
Hello! Sorry for the messy code! It seems that the website is mixing both the Raw and English comics in their latest updates. I noticed that there's a separate section for the Raw comics on their website, so I wrote that filter to exclude the Raw comics unless they are specifically searched for.

Is it not allowed to do that? Thank you